gradle/findbugs-exclude.xml

```diff
@@ -15,14 +15,92 @@
    limitations under the License.
 -->
 
+<!-- Findbugs filtering.
+
+Findbugs is a static code analysis tool run as part of the "check" phase of the build.
+This file dictates which categories of bugs and individual false positives that we supress.
+
+Findbugs tends to work a little bit better with Java than with Scala.  We
+suppress a few extra categories of bug reports when using Scala, since findbugs
+gives too many false positives in some cases.
+
+For a detailed description of findbugs bug categories, see http://findbugs.sourceforge.net/bugDescriptions.html
+-->
 <FindBugsFilter>
-    <!-- Exclude a few findbugs codes.
-         EI: May expose internal representation by returning reference to mutable object.
-         EI2: May expose internal representation by incorporating reference to mutable object.
-         MS: Malicious code vulnerability.
-         See http://findbugs.sourceforge.net/bugDescriptions.html for a full description of the findbugs codes.
-    -->
-    <Match>
-        <Bug code="EI,EI2,MS"/>
+    <Match>
+        <!-- Disable warnings about mutable objects and the use of public fields.
+            EI_EXPOSE_REP: May expose internal representation by returning reference to mutable object
+            EI_EXPOSE_REP2: May expose internal representation by incorporating reference to mutable object
+            MS_PKGPROTECT: Field should be package protected -->
+        <Bug pattern="EI_EXPOSE_REP,EI_EXPOSE_REP2,MS_PKGPROTECT"/>
+    </Match>
+
+    <Match>
+        <!-- Disable warnings about System.exit, until we decide to stop using it.
+            DM_EXIT: Method invokes System.exit -->
+        <Bug pattern="DM_EXIT"/>
+    </Match>
+
+    <Match>
+        <!-- Disable warnings about the lack of equals() when compareTo() is implemented.
+            EQ_COMPARETO_USE_OBJECT_EQUALS: This class defines a compareTo method but no equals() method -->
+        <Bug pattern="EQ_COMPARETO_USE_OBJECT_EQUALS"/>
+    </Match>
+
+    <Match>
+        <!-- These warnings generate huge numbers of false positives in Scala, so disable them when
+             processing Scala code.
+
+            NP_LOAD_OF_KNOWN_NULL_VALUE: The variable referenced at this point is known to be null
+            due to an earlier check against null.
+            NP_NULL_PARAM_DEREF: Method call passes null for non-null parameter.
+            NP_NULL_ON_SOME_PATH: Possible null pointer dereference
+            SE_BAD_FIELD: Non-transient non-serializable instance field in serializable class.
+            DM_STRING_CTOR: Method invokes inefficient new String(String) constructor.
```

Comment:
Yes, I'm aware of how findBugs works and that it flags issues in the code generated by the Scala compiler. Sorry for not making that clear. There were a few new ones that I didn't come across last time I did this exercise and I was curious about the code pattern that triggered in case you had already tracked it down. No worries.